src/controllers/user.controller.ts

```diff
@@ -95,9 +99,85 @@ const userController = new Elysia()
           email: t.Optional(t.String())
         })
       })
+      .post("/update-avatar", async ({ user, body }) => {
+        return await userService.updateAvatar(user as AuthUser, body.avatar ?? "")
+      }, {
+        checkAuth: ['user'],
+        detail: {
+          tags: ["User"],
+          security: [
+            { JwtAuth: [] }
+          ],
+        },
+        body: t.Object({
+          avatar: t.Optional(t.String()),
+          fullName: t.Optional(t.String()),
+          email: t.Optional(t.String())
+        })
+      })
+      .post("/2fa/setup", async ({ user }) => {
+        return await twoFactorService.setup2FA(user.id)
+      }, {
+        checkAuth: ['user'],
+        detail: {
+          tags: ["2FA"],
+          summary: "Setup 2FA for user",
+          description: "Generate 2FA secret and QR code for user to scan",
+          security: [
+            { JwtAuth: [] }
+          ],
+        },
+      })
+      .post("/2fa/enable", async ({ user, body }) => {
+        const success = await twoFactorService.enable2FA(user.id, body.token)
+        if (!success) {
+          throw new Error("Invalid 2FA token")
+        }
+        return { message: "2FA enabled successfully" }
+      }, {
+        checkAuth: ['user'],
+        detail: {
+          tags: ["2FA"],
+          summary: "Enable 2FA for user",
+          description: "Enable 2FA after verifying the token from authenticator app",
+          security: [
+            { JwtAuth: [] }
+          ],
+        },
+        body: t.Object({
+          token: t.String(),
+        })
+      })
+      .post("/2fa/disable", async ({ user }) => {
+        await twoFactorService.disable2FA(user.id)
+        return { message: "2FA disabled successfully" }
+      }, {
+        checkAuth: ['user'],
+        detail: {
+          tags: ["2FA"],
+          summary: "Disable 2FA for user",
+          description: "Disable 2FA and remove secret",
+          security: [
+            { JwtAuth: [] }
+          ],
+        },
+      })
+      .get("/2fa/status", async ({ user }) => {
+        return await twoFactorService.get2FAStatus(user.id)
+      }, {
+        checkAuth: ['user'],
+        detail: {
+          tags: ["2FA"],
+          summary: "Get 2FA status",
+          description: "Check if 2FA is enabled for the user",
+          security: [
+            { JwtAuth: [] }
+          ],
+        },
+      })
 
       .put("/update-password", async ({ user, body }) => {
```

Comment:
where is verify 2fa token when after user login success so they can call ? even in login follow you need to check if 2 fa enabled we dont return jwt  but return something like nextStep: 2fa